### PR TITLE
Hybrasyl Server 0.7.0

### PR DESCRIPTION
**Pull request by baughj**
## Description
Release PR for Hybrasyl Server 0.7.0 ("Daithi"). 

The next release of Hybrasyl will occur on or before March 15.

### Bug Fixes

* Items that are generated by scripting or other events can now be picked up correctly.
* Coma handling in groups now works as you would expect.
* Creature death no longer results in 0-gold piles.
* EndDialog scripting function should not only end dialog, but also close open dialog windows
* Correct bug in spawn calculations where it would take longer for initial spawns to populate than usual
* Fix several bugs in grouping (player death would be immediate, etc)
* Fix a number of dialog and NPC bugs (sequences would terminate inappropriately; NPC would lose state)

### Features

* Reactors (map tiles that can react to nearby events, such as dropped
  items or a player or monster moving into the tile) implemented.
* Full spawning implementation. SpawnGroups, Spawns now spawn and populate correctly with loot/xp/gold.
* Loot implemented (Loot tables, loot sets, etc). New `LootBox` class for handling loot calculations.
* Castables can now use and fire Lua scripts and start dialog sequences.
* JumpDialogs and FunctionDialogs added. Jump dialogs can be used in
  an NPC dialog tree to start new sequences or jump to different
  dialogs. Function dialogs can be used to run arbitrary Lua scripting
  code in between two dialogs (potentially to take an action - such as
  spawning a monster, dropping an item, starting a dialog, etc).
* Add two new dialog types for ease of scriptable dialog construction (TextAndJump, SimpleDialogSequence)
* Asynchronous dialog support (dialogs between two players, or a player and an NPC) implemented.
* Global dialog support implemented (reusable, scriptable dialogs)
* Scripting support improved and extended as was needed to support various beginning game NPCs.
* Hybrasyl time system reworked a bit (simplified). Legend marks
  updated to support "timestamp" and "creation" date; timestamp being
  what is displayed in the legend.
* Significant functionality improvements to scripting API. NPCs can now have temporary "memory" they can use to remember general information. 
* Add support to transition between two active dialog sequences.
* Add support for dialog options to trigger new sequences.
* Implement TakeItem for scripts, so that NPCs can remove items from a player's inventory
* Add support for setting and clearing permanent and session cookies (arbitrary player information which can be used by scripts).
* XML/XSD overhaul. Duplicate types removed, many things cleaned up & fixed. XML objects and typing are in a much better, cleaner state than previously. No impact to end user XML besides some namespacing changes.

### Technology / System Improvements

* Hybrasyl upgraded to .NET Core. As a result, Hybrasyl Server can now run on GNU/Linux and OSX, among others.
* MoonScript now used for Lua runtime instead of nLua.
* Serilog now used for logging (log4net replaced). Logging is now much cleaner. Game activity / scripting / spawning now have separate logfiles.
* Upgrade to xsd2code 5.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,21 +78,39 @@ at runtime; XML is processed when the server starts up for actual world data
...
+
+   Examine the Hybrasyl configuration in the Hybrasyl data directory, `config.xml`. In particular,
+   you will want to add the name of your character to `<Privileged>`, which will allow
+   them to use any slash command.
+
+4. Install and run Hybrasyl (see _Running Hybrasyl_ below).
+
+## Running Hybrasyl
+
+Hybrasyl Server is .NET Core, which means it can be run on a variety of platfoms (Windows, GNU/Linux, OSX).
```
Comment (norrismiv) [inline README.md]: platforms*

File: README.md [hunk 2]
```diff
@@ -72,21 +78,39 @@ at runtime; XML is processed when the server starts up for actual world data
...
+A `systemd` unit file [is provided](./contrib/hybrasyl.unit) to start the server on Ubuntu 18.04+. In any case,
+[download the latest release](https://github.com/hybrasyl/server/releases) for your platform. This can be unpacked
+into `/srv/hybrasyl` on GNU/Linux or a directory of your choosing on Windows.
+
+Once downloaded, either run the server directly (`Hybrasyl.exe` or
+`Hybrasyl` on GNU/Linux) or, if you’re running on GNU/Linux or a WSL
+distribution that uses systemd, install the unit file in
+`/etc/systemd/system/hybrasyl.service` and start Hybrasyl:
+
+`service hybrasyl start`
```
Comment (norrismiv) [inline README.md]: may want to include `dotnet run` commands for running it as a non-service on osx/linux

File: README.md [hunk 3]
```diff
@@ -95,29 +119,22 @@ The process for compiling Hybrasyl is detailed below.
...
+
 2. Clone the [launcher](https://github.com/hybrasyl/launcher) and 
    [server](https://github.com/hybrasyl/server) repositories to your local machine
    using a [git client](https://git-scm.com/downloads/guis), or with Visual Studio's built-in integration. 
    Make sure you clone them into separate directories. 
+
 3. Open the Hybrasyl Server solution (`Hybrasyl.sln`) in Visual Studio and
    update all NuGet packages (just building it will do this). The SDK for
    XML is now included in server to make this process (as well as making changes) 
    easier.
```
Comment (norrismiv) [inline README.md]: alternatively if running `dotnet` commands you can use `dotnet restore` in the same directory as the `.csproj` file.

File: README.md (hunk 2 above)
Comment (baughj) [reply README.md]: My intent is to publish artifacts so you don't need dotnet run; the build process builds as ELF and can be run from command line normally (literally just `./Hybrasyl`)

File: README.md (hunk 1 above)
Comment (baughj) [reply README.md]: lol. fixed

File: README.md (hunk 2 above)
Comment (norrismiv) [reply README.md]: sure, but in the cast of debug/test on osx/linux the run commands will help

File: README.md (hunk 3 above)
Comment (baughj) [reply README.md]: added

File: README.md (hunk 2 above)
Comment (baughj) [reply README.md]: 👍 - will add

**Comment by baughj** [comment]
🎉

**Comment by baughj** [comment]
A little late due to the neverending dumpster fire that is 2020, but, it is here

**Pull request merged**